Makefile

```diff
@@ -25,7 +25,7 @@ ADDONS_DIR := addons
 YTT_TESTS_DIR := pkg/v1/providers/tests
 PACKAGES_SCRIPTS_DIR := $(abspath hack/packages/scripts)
 UI_DIR := tkg/web
-GO_MODULES=$(shell find . -path "*/go.mod" | grep -v "^./pinniped" | xargs -I _ dirname _)
+GO_MODULES=$(shell find . -path "*/go.mod" | grep -v "^./pinniped$$" | xargs -I _ dirname _)
```

Comment:
What is this trying to achieve?  IIUC it is trying grep out the `pinniped` modules while leaving in "pinniped-components"?

I don't think that works because the `grep -v` is being run on the output that still contains the `go.mod` suffix.  If I run the shell command manually on my shell, I get all the `pinniped/...` directories left too.

How about grepping out `"^./pinniped/"` (with the `/` at the end)?

Reply:
i'm seeing a different result:

```
➜ find . -path "*/go.mod" | grep -v "^./pinniped$"
./addons/hack/tools/go.mod
./addons/go.mod
./apis/cli/go.mod
./apis/config/go.mod
./apis/run/go.mod
./apis/addonconfigs/go.mod
./apis/core/go.mod
./cmd/cli/plugin-admin/builder/go.mod
./cmd/cli/plugin-admin/codegen/go.mod
./cmd/cli/plugin-admin/test/go.mod
./cmd/cli/plugin/cluster/go.mod
./cmd/cli/plugin/login/go.mod
./cmd/cli/plugin/managementcluster/go.mod
./cmd/cli/plugin/package/go.mod
./cmd/cli/plugin/pinniped-auth/go.mod
./cmd/cli/plugin/feature/go.mod
./cmd/cli/plugin/secret/go.mod
./hack/tools/go.mod
./hack/packages/kbld-image-replace/go.mod
./hack/packages/package-tools/go.mod
./pkg/v1/providers/tests/go.mod
./test/pkg/go.mod
./packages/cluster-api-bootstrap-kubeadm/test/go.mod
./packages/cluster-api-provider-aws/test/go.mod
./packages/cluster-api-provider-azure/test/go.mod
./packages/cluster-api-provider-docker/test/go.mod
./packages/cluster-api-provider-vsphere/test/go.mod
./packages/cluster-api/test/go.mod
./cli/runtime/hack/tools/go.mod
./cli/runtime/go.mod
./cli/core/hack/tools/go.mod
./cli/core/go.mod
./tkg/hack/tools/go.mod
./tkg/go.mod
./capabilities/client/go.mod
./capabilities/controller/go.mod
./featuregates/client/go.mod
./featuregates/controller/go.mod
./packageclients/go.mod
./pinniped-components/hack/tools/go.mod
./pinniped-components/post-deploy/go.mod
./pinniped-components/tanzu-auth-controller-manager/go.mod
./go.mod
```

Reply:
But i think yours is more correct.